mcs/class/Facades/System.Runtime.InteropServices.RuntimeInformation/System.Runtime.InteropServices/RuntimeInformation.cs

```diff
@@ -35,12 +35,14 @@ namespace System.Runtime.InteropServices
 {
 	public static class RuntimeInformation
 	{
-		[DllImport ("__Internal")]
-		extern static string mono_get_runtime_build_info ();
+		static string frameworkDescription;
 
 		public static string FrameworkDescription {
 			get {
-				return mono_get_runtime_build_info ();
+				if (frameworkDescription == null)
+					frameworkDescription = "Mono " + (string)Type.GetType ("Mono.Runtime").GetMethod ("GetDisplayName", BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic).Invoke (null, null);
```

Comment:
This is not good solution as it breaks linker for example. Better to do InternalsVisibleTo from mscorlib or add new icall